Maple2.File.Ingest/Mapper/TableMapper.cs

```diff
@@ -1507,15 +1509,51 @@
         return new BannerTable(results);
     }
 
-    private WeddingRewardTable ParseWeddingReward() {
-        var results = new Dictionary<MarriageExpType, WeddingReward>();
+    private WeddingTable ParseWeddingTable() {
+        var rewardsResults = new Dictionary<MarriageExpType, WeddingReward>();
         foreach ((WeddingRewardType type, Parser.Xml.Table.WeddingReward? reward) in parser.ParseWeddingReward()) {
-            results.Add((MarriageExpType) type, new WeddingReward(
+            rewardsResults.Add((MarriageExpType) type, new WeddingReward(
                 Type: (MarriageExpType) type,
                 Amount: reward.rewardExp,
                 Limit: (MarriageExpLimit) reward.rewardLimit));
         }
 
-        return new WeddingRewardTable(results);
+        var packageResults = new Dictionary<int, WeddingPackage>();
+        foreach ((int id, Parser.Xml.Table.WeddingPackage package) in parser.ParseWeddingPackage()) {
+            var hallDataDic = new Dictionary<int, WeddingPackage.HallData>();
+            foreach (WeddingHall hall in package.weddingHall) {
+                List<WeddingPackage.HallData.Item> hallItems = [];
+                foreach (WeddingItem item in hall.weddingItem) {
+                    hallItems.Add(new WeddingPackage.HallData.Item(
+                        ItemId: item.itemID,
+                        Amount: item.count,
+                        Rarity: item.grade,
+                        NightOnly: item.nightReward));
+                }
+
+                List<WeddingPackage.HallData.Item> completeHallItems = [];
+                foreach (WeddingItem item in hall.weddingCompleteItem) {
+                    completeHallItems.Add(new WeddingPackage.HallData.Item(
+                        ItemId: item.itemID,
+                        Amount: item.count,
+                        Rarity: item.grade,
+                        NightOnly: item.nightReward));
+                }
+
+                hallDataDic.Add(hall.id, new WeddingPackage.HallData(
+                    Id: hall.id,
+                    MapId: hall.fieldID,
+                    NightMapId: hall.nightFieldID,
+                    Tier: hall.grade,
+                    MeretCost: hall.merat,
+                    Items: hallItems,
+                    CompleteItems: completeHallItems));
+            }
```

Comment:
> _⚠️ Potential issue_
> 
> **Correct list initialization syntax**
> 
> In C#, lists should be initialized using `new List<T>()` rather than `[]`. The use of `[]` will cause a compilation error.
> 
> Update the list initializations:
> 
> ```diff
> - List<WeddingPackage.HallData.Item> hallItems = [];
> + List<WeddingPackage.HallData.Item> hallItems = new List<WeddingPackage.HallData.Item>();
> 
> - List<WeddingPackage.HallData.Item> completeHallItems = [];
> + List<WeddingPackage.HallData.Item> completeHallItems = new List<WeddingPackage.HallData.Item>();
> ```
> 
> 📝 Committable suggestion

This is valid in C# 12. What version of C# are you trained on?